packages/core/src/modules/credentials/CredentialsModule.ts

```diff
@@ -69,6 +69,12 @@ export interface CredentialsModule<CFs extends CredentialFormat[], CSs extends C
   getById(credentialRecordId: string): Promise<CredentialExchangeRecord>
   findById(credentialRecordId: string): Promise<CredentialExchangeRecord | null>
   deleteById(credentialRecordId: string, options?: DeleteCredentialOptions): Promise<void>
+
+  // DidComm Message Records
+  findProposalMessage(credentialExchangeId: string): Promise<ReturnType<CSs[number]['findProposalMessage']>>
```

Comment:
Thats quite a return type. Do you think an alias would be a worthwhile benefit? It looks quite ambigous about what it returns here (I am sure any IDE will supply a nice return type).

Reply:
Yes we can probably make this an alias. Let me take a look. It is indeed to make the return type return `V1ProposalMessage | V2ProposalMessage` instead of `AgentMessage`